torch/csrc/api/include/torch/nn/modules/embedding.h

```diff
@@ -35,14 +69,39 @@ class TORCH_API EmbeddingImpl : public torch::nn::Cloneable<EmbeddingImpl> {
   /// `indices` supplied and returns the result.
   Tensor forward(const Tensor& indices);
 
+  static EmbeddingImpl& from_pretrained(Tensor embeddings, bool freeze = true, c10::optional<int64_t> padding_idx = c10::nullopt,
+          c10::optional<float> max_norm = c10::nullopt, float norm_type = 2., bool scale_grad_by_freq = false, bool sparse = false);
```

Comment:
Sorry that I think we will need to move `from_pretrained` to the `Embedding` class, not `EmbeddingImpl`, because we want people to be able to use it with `torch::nn::Embedding::from_pretrained(...)`. I wrote a gist to illustrate the idea: https://gist.github.com/yf225/8eee0ef3f6afd2317092900927a43994.

Reply:
We should also use `EmbeddingOptions` instead of writing out the parameters explicitly, as shown in the gist.

Reply:
Also the return type shouldn't be a reference, and should just be `Embedding` (after we move this to the `Embedding` class).